### PR TITLE
Updated backend Z3 to work with the newest Z3 version

### PR DESCRIPTION
**Pull request by Phat3**
Signed-off-by: Sebastiano Mariani <mariani.sebastiano@gmail.com>

### CODE REVIEW

File: claripy/backends/backend_z3.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
 import os
 import z3
+import gc
```
Comment (ltfish) [inline claripy/backends/backend_z3.py]: Why do we need `gc`?

File: claripy/backends/backend_z3.py [hunk 2]
```diff
@@ -46,6 +47,14 @@ def _add_memory_pressure(p):
 
 supports_fp = hasattr(z3, 'fpEQ')
 
+# Hack to get a reference to z3libs. Thanks Rhelmot!
```
Comment (ltfish) [inline claripy/backends/backend_z3.py]: `Rhelmot` -> `rhelmot`

@rhelmot Correct me if I'm wrong.

**Comment by ltfish** [comment]
Thanks for doing this! But I feel you are committing some local workarounds, which should not be part of this patch. Can you please check this PR again?

File: claripy/backends/backend_z3.py (hunk 1 above)
Comment (rhelmot) [reply claripy/backends/backend_z3.py]: @Phat3 you can get rid of the gc hack and replace the usage of the libz3 object with this: https://github.com/Z3Prover/z3/pull/2143

**Pull request merged**